projects/packages/woocommerce-analytics/src/class-woo-analytics-trait.php

```diff
@@ -816,7 +823,7 @@ public function get_session_cookie() {
 	 * @return string
 	 */
 	public function get_current_url() {
-		return sanitize_url( wp_unslash( ( empty( $_SERVER['HTTPS'] ) ? 'http' : 'https' ) . "://$_SERVER[HTTP_HOST]$_SERVER[REQUEST_URI]" ) ); // phpcs:ignore WordPress.Security.ValidatedSanitizedInput.InputNotValidatedNotSanitized -- actually escaped with sanitize_url.
+		return home_url( add_query_arg( null, null ) );
```

Comment:
I replaced usage of the `$_SERVER` variable with `home_url()` to retrieve the current URL, as `home_url()` should generally be more dependable. 

No extra sanitization was applied, IMO, the method's purpose is to return the raw current URL, and leave any necessary sanitization to the consumer. 

For example, we don't need to sanitize the URL when passing it to event properties, since we’re not outputting it or storing it in the database. We're just sending it to remote servers and the server is responsible for sanitizing it.